src/libs/nav.ts

```diff
@@ -7,6 +7,38 @@ import { useOnMount, useStore } from 'src/libs/react-utils';
 import { routeHandlersStore } from 'src/libs/state';
 import * as Utils from 'src/libs/utils';
 
+// TODO: add all used nav key names here and switch consumers to use the safe const values here
+/**
+ * union type (enum-like) of safe nav key names
+ */
+export type TerraNavKey = 'billing' | 'workspace-dashboard' | 'workspaces';
+
+export type TerraNavKeyLookup = { [key in TerraNavKey]: key };
+
+// TODO: fill out this value lookup to reflect full list of key names once available above
+/**
+ * allowed values for nav keys in a lookup table
+ * Note: use terraNavKey type guard method below instead of this const.
+ *   Typescript currently allows unsafe object['not-a-enum-key-name'] indexing
+ *   lookups into this otherwise type-safe construct.
+ */
+export const terraNavKeyValue: TerraNavKeyLookup = {
+  billing: 'billing',
+  'workspace-dashboard': 'workspace-dashboard',
+  workspaces: 'workspaces',
+};
+
+/**
+ * type guard for nav keys to better enforce key names with snake-case
+ * @param key
+ */
+export const terraNavKey = (key: TerraNavKey): TerraNavKey => key;
```

Comment:
this extra type guard fn is needed because 'the-nav-key', devs can't use dot notation to access the enum-constants, so this pass-thru helps provide the same safety.

Reply:
`enforceType` sounds good to me.

Here's the relevant TS docs: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

An example usage might help here, showing how this can be used to require a narrower type than what the function that its return value will be passed to requires.

Reply:
fixed